modules/optimization/test/tests/optimizationreporter/optimizationdata/tests

```diff
@@ -0,0 +1,15 @@
+[Tests]
+  issues = '#24333'
+  design = 'OptimizationData.md'
+  [readData]
+    type = CSVDiff
+    max_threads = 1 # Optimize executioner does not support multiple threads
```

Comment:
It is not, it is only testing the optimimizationData reporter to see if it can sample two variables correctly and then correctly scale the simulation_values by the weights